File: src/api-umbrella/admin-ui/app/models/api/rewrite.js
import Model, { attr } from '@ember-data/model';
import { t } from 'api-umbrella-admin-ui/utils/i18n';
import { buildValidations, validator } from 'ember-cp-validations';

const Validations = buildValidations({
  matcherType: [
    validator('presence', {
      presence: true,
      description: t('Matcher Type'),
    }),
  ],
  httpMethod: [
    validator('presence', {
      presence: true,
      description: t('HTTP Method'),
    }),
  ],
  frontendMatcher: [
    validator('presence', {
      presence: true,
      description: t('Frontend Matcher'),
    }),
  ],
  backendReplacement: [
    validator('presence', {
      presence: true,
      description: t('Backend Replacement'),
    }),
  ],
});

class Rewrite extends Model.extend(Validations) {
  static validationClass = Validations;

  @attr('number')
  sortOrder;

  @attr()
  matcherType;

  @attr()
  httpMethod;

  @attr()
  frontendMatcher;

  @attr()
  backendReplacement;
}

export default Rewrite;

File: test/support/models/api/rewrite.rb
class Api::Rewrite
  include Mongoid::Document
  field :_id, :type => String, :overwrite => true, :default => lambda { SecureRandom.uuid }
  field :matcher_type, :type => String
  field :http_method, :type => String
  field :frontend_matcher, :type => String
  field :backend_replacement, :type => String
  embedded_in :api
end
